### PR TITLE
Zh cn/update documentations, success stories, ruby from other languages pages

### PR DESCRIPTION
**Pull request by 1ternal**
Translate the Documentation page  (zh_cn)

Related #1074

### CODE REVIEW

**Comment by JuanitoFatas** [comment]
Nice! Next time please make a smaller PR, easier for people to review, thank you!

**Comment by JuanitoFatas** [comment]
@pmq20 @hlcfan @liuzxc @tzwm @happybai @diguage @alex-marmot @itsliupeng
@cassiuschen @1ternal @killernova @VincentZhao @reyesyang @jun1st @springwq
@Huibean @junzh0u @zigzagpig @42thcoder @GaaraLi @shouxian @johnnyhappy365
@aidewoode @Yunfei1982 @Airmacho

Please review this pull request! :bow: :smiley:

File: zh_cn/documentation/index.md [hunk 1]
```diff
@@ -6,11 +6,20 @@ lang: zh_cn
 
 在这里您可以找到手册、教程和参考资料的链接，在你喜欢上 Ruby 编程的时候，他们将帮上大忙。
 
+### 安装Ruby
+
+除非只想在浏览器里尝试 Ruby (查看下面的链接),否则你的计算机需要安装 Ruby。打开终端，输入以下命令检查 Ruby 是否可用
```
Comment (hlcfan) [inline zh_cn/documentation/index.md]: Comma.

File: zh_cn/documentation/index.md [hunk 2]
```diff
@@ -26,34 +35,34 @@ lang: zh_cn
 : 一本不同寻常但是非常有趣的书，通过故事、幽默和漫画来教你学习 Ruby,
   由 *why the luckystiff* 创作, 本书仍然是学习 Ruby 的经典之作。
 
-[花20分钟体验](/zh_cn/documentation/quickstart/)
+[Ruby 20分钟入门](/zh_cn/documentation/quickstart/)
```
Comment (hlcfan) [inline zh_cn/documentation/index.md]: :+1:

File: zh_cn/documentation/index.md [hunk 3]
```diff
@@ -26,34 +35,34 @@ lang: zh_cn
...
 
-[Learn to Program][8]
-: 由 Chris Pine 编写的编程入门，如果你还不了解程序，从这里开始吧。
+[学习编程][8]
+: 由 Chris Pine 为编程入门者编写的非常棒的小教程。如果你不知道如何编程，从这里开始。
 
-[Learn Ruby the Hard Way][38]
-: 一系列非常好的带有说明的练习，可以指导你从基本的 Ruby 一路进阶到 OOP 和 WEB 开发。
+[笨方法学 Ruby][38]
+: 一系列非常好的带有说明的练习，可以指导你从基本的 Ruby 一路进阶到 面向对象编程 和 网页 开发。
```
Comment (hlcfan) [inline zh_cn/documentation/index.md]: No spaces needed here.
**一路进阶到 面向对象编程和网页开发**

File: zh_cn/documentation/index.md [hunk 4]
```diff
@@ -67,18 +76,17 @@ lang: zh_cn
...
 [APIdock][18]
-: 带有用户评注的 Ruby， Rails 和 RSpec 文档。
+: 带有用户评注的 Ruby、Rails 和 RSpec 文档。
 
-### 编辑器和IDE
+### 编辑器和集成开发环境
 
-您可以用您操作系统上默认的编辑器来进行Ruby编程，为了更高效的编程，选择一个基本特性支持 Ruby （比如 代码高亮，文件浏览）的源代码编辑器或者
-对Ruby有着高级特性支持（如 代码补全，重构，测试等） 的集成开发环境（IDE）是非常值得的。
+您可以用您操作系统上默认的编辑器来进行 Ruby 编程。此外，为了更高效的编程，选择一个支持 Ruby 基本特性（比如 代码高亮，文件浏览）的代码编辑器或者对 Ruby 有着高级特性支持（比如 代码补全，重构，测试支持等）的集成开发环境是非常值得的。
```
Comment (hlcfan) [inline zh_cn/documentation/index.md]: I thought you've changed the original meaning here, please put it back to
**选择一个支持基本特性支持 Ruby（比如 代码高亮，文件浏览）的代码编辑器**

File: zh_cn/documentation/ruby-from-other-languages/index.md [hunk 5]
```diff
@@ -1,16 +1,13 @@
...
 ---
 
-{% include out-of-date.html %}
-
-
 当您初次看到 Ruby 语言代码的时候，它很可能让您想起您使用过的其他编程语言。这是当然的， 因为 Ruby 的大部分语法是
 Perl、Python 和 Java（以及其他语言）的用户所熟悉的。 如果您曾经使用过这这些语言，学习 Ruby 就是小菜一碟。
 
-本文包含了两部分。第一部分说的是从其他语言到 Ruby 时的一些小窍门； 第二部分介绍了 Ruby 语言和其他语言相比较时的一些特性。
+本文包含了两部分。第一部分简明扼要的概述了当你从 *X* 到 Ruby 语言的一些所见所闻； 第二部分介绍了 Ruby 语言主要特性以及与你所熟悉的语言之间的差异。
```
Comment (hlcfan) [inline zh_cn/documentation/ruby-from-other-languages/index.md]: 简明扼要**地**概述

File: zh_cn/documentation/ruby-from-other-languages/index.md [hunk 6]
```diff
@@ -37,12 +34,12 @@ Java），或者以列表做循环（比如 Perl 的 `for (@a) {...}`， 还有
 
 {% highlight ruby %}
 some_list.each do |this_item|
-  # We're inside the block.
-  # deal with this_item.
+  # 我们现在在代码块内.
```
Comment (hlcfan) [inline zh_cn/documentation/ruby-from-other-languages/index.md]: Period.

**Comment by 1ternal** [comment]
I got this, thx @hlcfan

File: zh_cn/documentation/ruby-from-other-languages/index.md [hunk 7]
```diff
@@ -113,46 +110,48 @@ end
...
+### 关键字参数
 
-Ruby 并不像 Python 一样支持命名参数，但可以用符和字典来替代。 Ruby on Rails 和其他的一些 Ruby
-程序都使用了这种方法。例如：
+类似 Python, 从 Ruby 2.0 开始，可以使用关键字参数定义方法
 
 {% highlight ruby %}
-def some_keyword_params( params )
-  params
+def deliver(from: "张三", to: nil, via: "邮件")
```
Comment (hlcfan) [inline zh_cn/documentation/ruby-from-other-languages/index.md]: Using Chinese chars as params looks kinda weird to me.

File: zh_cn/documentation/ruby-from-other-languages/index.md [hunk 8]
```diff
@@ -232,150 +229,128 @@ NoMethodError: private method `func' called for #<Test:0x342784>
...
 
-# 14 hours from 00:00 January 1st
-# (aka when you finally wake up ;)
+# 一月一号之后的 14 个小时
+# (也就是你醒来的时间)
 Time.mktime(2006, 01, 01) + 14.hours # => Sun Jan 01 14:00:00
 {% endhighlight %}
 
-### Funny method names
+### 有趣的方法名
```
Comment (hlcfan) [inline zh_cn/documentation/ruby-from-other-languages/index.md]: What about 有趣的方法

File: zh_cn/documentation/ruby-from-other-languages/index.md [hunk 9]
```diff
@@ -232,150 +229,128 @@ NoMethodError: private method `func' called for #<Test:0x342784>
...
 
-Singleton methods are per-object methods. They are only available on the
-Object you defined it on.
+单例方法是单一对象方法。只在定义过的对象上可用。
 
 {% highlight ruby %}
 class Car
   def inspect
-    "Cheap car"
+    "屌丝车 (╯‵□′)╯︵┻━┻"
```
Comment (hlcfan) [inline zh_cn/documentation/ruby-from-other-languages/index.md]: :+1:

Comment (springwq) [inline zh_cn/documentation/ruby-from-other-languages/index.md]: I don't think "屌丝" is a good word for an official site.

Comment (hlcfan) [inline zh_cn/documentation/ruby-from-other-languages/index.md]: @springwq YOU ABSOLUTELY RIGHT!

File: zh_cn/documentation/ruby-from-other-languages/index.md (hunk 8 above)
Comment (1ternal) [inline zh_cn/documentation/ruby-from-other-languages/index.md]: I thought this part is focused on method's name, not method itself. :cry: 
btw I'm totally noob. thanks for your help.

File: zh_cn/documentation/ruby-from-other-languages/index.md (hunk 9 above)
Comment (1ternal) [inline zh_cn/documentation/ruby-from-other-languages/index.md]: sorry for 三观不正 :sob:

File: zh_cn/documentation/ruby-from-other-languages/to-ruby-from-c-and-cpp/index.md [hunk 10]
```diff
@@ -1,7 +1,89 @@
...
 layout: page
-title: "从 C/C++ 到 Ruby - Ruby 官方网站"
+title: "从 C/C++ 到 Ruby"
 lang: zh_cn
 ---
 
-{% include out-of-date.html %}
+很难用一个列表来描述 Ruby 与 C 或 C++ 代码的不同之处，因为差异相当大。其中一个原因是 Ruby 的运行时帮你做了很多工作。与 C 的“不隐藏内部机制”原则不同——Ruby 的关注点在于将繁重的工作交由运行时，使用者站在运行时的肩上，活的更轻松。除非或直到做代码优化，使用 Ruby 不用费心“让编译器高兴” 。
+
+换句话说，Ruby 代码执行速度远远慢于“等价”的 C 或 C++ 代码是确定无疑的。相同的时间内，你头脑里关注的是如何快速的将一个Ruby程序运行起来，而且只需要很少的代码就可以做到。和把你搅成一锅粥的 C++ 相比，Ruby 要简单的多。
```
Comment (hlcfan) [inline zh_cn/documentation/ruby-from-other-languages/to-ruby-from-c-and-cpp/index.md]: **一个 Ruby 程序**

File: zh_cn/documentation/ruby-from-other-languages/to-ruby-from-c-and-cpp/index.md [hunk 11]
```diff
@@ -1,7 +1,89 @@
...
+
+Ruby 是动态类型语言，而不是静态类型。它在运行的时候运行时做尽可能多的事。例如，你不用知道 Ruby 程序“链接”（也就是 载入和使用）到哪里或之前调用什么方法。
```
Comment (hlcfan) [inline zh_cn/documentation/ruby-from-other-languages/to-ruby-from-c-and-cpp/index.md]: Please get rid of space **也就是 载入和使用**

File: zh_cn/documentation/ruby-from-other-languages/to-ruby-from-c-and-cpp/index.md [hunk 12]
```diff
@@ -1,7 +1,89 @@
...
+
+可喜的是,事实证明 Ruby 和 C 之间的共生关系很和谐。Ruby 支持所谓“扩展模块”。这些模块由 C 语言编写，可以被 Ruby 程序调用（从外部来说，与其他 Ruby 模块表现和行为毫无二致）。通过这种方式,可以将性能关键部分从Ruby软件中切分出来,再由 C 语言打造。
```
Comment (hlcfan) [inline zh_cn/documentation/ruby-from-other-languages/to-ruby-from-c-and-cpp/index.md]: **从 Ruby 软件**

File: zh_cn/documentation/ruby-from-other-languages/to-ruby-from-c-and-cpp/index.md [hunk 13]
```diff
@@ -1,7 +1,89 @@
...
+* 字符串使用双引号。
+* 字符串是可变类型。
+* 与手册页面类似，使用 `ri` 命令，你可以在命令行窗口浏览大部分的文档。
+* 可以使用同样的命令行调试器。
+
+### 与 C++ 相似点
+
+Ruby 与 C++ 一样的地方……
+
+* 几乎一样的操作符（甚至 `::`）。`<<` 常用于向列表内追加元素。值得注意的是：Ruby 使用`.`，而不使用 `->`。
```
Comment (hlcfan) [inline zh_cn/documentation/ruby-from-other-languages/to-ruby-from-c-and-cpp/index.md]: You may need space between **使用** and **`.`**.

File: zh_cn/documentation/ruby-from-other-languages/to-ruby-from-java/index.md [hunk 14]
```diff
@@ -1,7 +1,47 @@
...
+* 变量名只是标签。它们没有相应的类型。
+* 没有类型声明。按需分配变量名，及时可用（如：`a = [1,2,3]` 而不是 `int[] a = {1,2,3};`）。
+* 没有显式转换。只需要调用方法。代码运行之前，单元测试应该告诉你出现异常。
+* 是 `foo = Foo.new("hi")`，而不是 `Foo foo = new Foo("hi")`。
+* 构造器总是命名为“initialize” 而不是类名称。
+* 作为接口的替代，你将获得“混入（mixins）”。
+* 相比 XML，倾向于使用 YAML。
+* `nil` 替代 `null`。
+* `==` and `equals()` are handled differently in Ruby. Use `==` when you
+  want to test equivalence in Ruby (`equals()` is Java). Use `equal?()`
```
Comment (hlcfan) [inline zh_cn/documentation/ruby-from-other-languages/to-ruby-from-java/index.md]: Remove this line please.

File: zh_cn/documentation/success-stories/index.md [hunk 15]
```diff
@@ -1,7 +1,72 @@
...
+
+* [Open Domain Server][6] 使用 Ruby 帮助人们使用动态域名解析客户端实时更新IP配置，达到映射到静态域的目的。
+
+#### 电讯
+
+* [朗讯][7]一个3G无线电话产品中使用了 Ruby。
+
+#### 系统管理
+
+* [Level 3 Communications][8] 公司的 Unix 容量和规划系统负责收集分散在全球超过 1700 台 Unix(Solaris 和   Linux) 服务器的性能统计数据， 它的中央数据收集部分是由 Ruby 编写的。
```
Comment (hlcfan) [inline zh_cn/documentation/success-stories/index.md]: **Solaris 和 Linux**

**Comment by 1ternal** [comment]
Hi @li-thy-um, I've fixed the issue, please review again.  much appreciated!

**Comment by stomar** [comment]
Please clean up history before merge and use commit messages that conform to our conventions, like e.g. `Translate installation page (zh_cn)`

**Comment by stomar** [comment]
Is this ready to merge?

If yes, fix commit message, e.g. into `Translate and update several documentation pages (zh_cn)`

**Comment by JuanitoFatas** [comment]
Merged manually as fae41cb. Thanks for your contribution @1ternal and thanks reviewer and @stomar for coordinating. :heart:

**Comment by stomar** [comment]
:+1:

**Pull request closed without merging**